arviz/tests/helpers.py

```diff
@@ -571,11 +573,63 @@ def pystan_version():
     """
     try:
         import pystan  # pylint: disable=import-error
+
+        version = int(pystan.__version__[0])
     except ImportError:
-        import stan as pystan  # pylint: disable=import-error
-    return int(pystan.__version__[0])
+        try:
+            import stan as pystan  # pylint: disable=import-error
+
+            version = int(pystan.__version__[0])
+        except ImportError:
+            version = None
+    return version
 
 
 def test_precompile_models(eight_schools_params, draws, chains):
     """Precompile model files."""
     load_cached_models(eight_schools_params, draws, chains)
+
+
+def importorskip(
+    modname: str, minversion: Optional[str] = None, reason: Optional[str] = None
+) -> Any:
+    """Imports and returns the requested module ``modname``. Doesn't allow skips on CI machine
+        borrowed and modified from ``pytest.importorskip``
+    :param str modname: the name of the module to import
+    :param str minversion: if given, the imported module's ``__version__``
+        attribute must be at least this minimal version, otherwise the test is
+        still skipped.
+    :param str reason: if given, this reason is shown as the message when the
+        module cannot be imported.
+    :returns: The imported module. This should be assigned to its canonical
+        name.
+    Example::
+        docutils = pytest.importorskip("docutils")
+    """
+    # ARVIZ_CI_MACHINE is True if tests run on CI, where ARVIZ_CI_MACHINE env variable exists
+    ARVIZ_CI_MACHINE = os.environ.get("ARVIZ_CI_MACHINE") is not None
+    if ARVIZ_CI_MACHINE:
+        import warnings
+
+        compile(modname, "", "eval")  # to catch syntaxerrors
+
+        with warnings.catch_warnings():
+            # make sure to ignore ImportWarnings that might happen because
+            # of existing directories with the same name we're trying to
+            # import but without a __init__.py file
+            warnings.simplefilter("ignore")
+            __import__(modname)

```

Comment:
is there any difference between these

    __import__(modname)
    mod = sys.modules[modname]

vs

    mod = importlib.import_module(modname)

Reply:
I think they should be mostly similar for our uses. I used the former as I copied the existing code in `pytest.importorksip`.

But I can use `importlib.import_module`. It might clear up the failures in CI to import `jax.random.PRNGKey` and `pyro.infer.Predictive`.


I can't seem to easily install `jaxlib` on Windows, so I can't test this out locally.